src/Compilers/Core/Portable/Diagnostic/ExternalFileLocation.cs

```diff
@@ -14,12 +14,13 @@ namespace Microsoft.CodeAnalysis
     internal sealed class ExternalFileLocation : Location, IEquatable<ExternalFileLocation?>
     {
         private readonly TextSpan _sourceSpan;
-        private readonly FileLinePositionSpan _lineSpan;
+        private readonly FileLinePositionSpan _lineSpan, _mappedLineSpan;
 
-        internal ExternalFileLocation(string filePath, TextSpan sourceSpan, LinePositionSpan lineSpan)
+        internal ExternalFileLocation(string filePath, TextSpan sourceSpan, LinePositionSpan lineSpan, string mappedFilePath, LinePositionSpan mappedLineSpan)
         {
             _sourceSpan = sourceSpan;
             _lineSpan = new FileLinePositionSpan(filePath, lineSpan);
+            _mappedLineSpan = new FileLinePositionSpan(mappedFilePath, mappedLineSpan, hasMappedPath: !PathUtilities.PathsEqual(filePath, mappedFilePath));
```

Comment:
is the hasMappedPath portion of this tested?

Reply:
There are some checks in the test added by @mavasani:
https://github.com/dotnet/roslyn/blob/114f08f3bf747f8b8dedfa5addf11b197687d383/src/Compilers/CSharp/Test/Syntax/Diagnostics/LocationsTests.cs#L529-L547

Reply:
Or were you asking for a deeper testing @CyrusNajmabadi?

Reply:
nope.  that's answers the question.